sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledHashJoinExec.scala

```diff
@@ -348,9 +351,12 @@ object GpuShuffledHashJoinExec extends Logging {
           (Left(singleBuildCb), bufferedStreamIter)
 
         } else { // Other cases without optimization
-          val safeIter = GpuSubPartitionHashJoin.safeIteratorFromSeq(Seq(firstBuildBatch)) ++
-            coalesceBuiltIter
-          val gpuBuildIter = if (isBuildSerialized) {
+          val safeIter = {
+            val autoCloseableSeq = Seq(firstBuildBatch.asInstanceOf[AutoCloseable])
```

Comment:
Since the iterator now returns AutoCloseable, we shouldn't need the cast.

```suggestion
            val autoCloseableSeq = Seq(firstBuildBatch)
```